## Reason to use Python NumPy Array instead of List

### 1. High Performance (Fast)

In [10]:
# Python List - Cube of number
%timeit  cube = [i**3 for i in range(100)]

63.9 µs ± 519 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)


In [11]:
#NumPy code for Cube of number
num = np.arange(100)
%timeit num**3

2.37 µs ± 68.8 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)


### Conclusion: 
List took 63.9 ms whereas the numpy array took just 2.37ms. Hence, numpy array is faster than list.

### 2. Less Memory

In [12]:
# List Memory allocation
import time
import sys
cube = [i**3 for i in range(100)]
print("Memory Allocated in List =",sys.getsizeof(cube)*len(cube))

Memory Allocated in List = 91200


In [13]:
# NumPy Memory allocation
import numpy as np 
num = np.arange(100)
cube = num**3
print("Memory Allocated in NumPy =",num.size*cube.itemsize)

Memory Allocated in NumPy = 400


### Conclusion: 
The difference between the list and numpy is very high and this makes python numpy array as the preferred choice over list.